sdv/data_processing/data_processor.py

```diff
@@ -751,7 +751,17 @@ def reverse_transform(self, data, reset_keys=False):
             if is_integer_dtype(dtype) and is_float_dtype(column_data.dtype):
                 column_data = column_data.round()
 
-            reversed_data[column_name] = column_data[column_data.notna()].astype(dtype)
+            try:
+                reversed_data[column_name] = column_data[column_data.notna()].astype(dtype)
```

Comment:
Can you move the `reversed_data[column_name] = column_data[column_data.notna()]` outside the `try`? So that we are sure the `ValueError` was raised due to the `astype`. You can then remove line 762 as well.

Reply:
Great tip 🙏🏻 thanks